examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp

```diff
@@ -47,6 +48,41 @@ using namespace ::chip::DeviceLayer;
 uint32_t identifyTimerCount;
 constexpr uint32_t kIdentifyTimerDelayMS = 250;
 
+namespace {
+
+void StartMdnsAdvertise(bool isIpv4)
+{
+#if !INET_CONFIG_ENABLE_IPV4
```

Comment:
in my reading of the code, this is unreachable, because clients of this API check the ifdef

what happens in this code if the device has both an ipv4 and an ipv6 address?  this function would get called twice, once with true and once with false, and the code on line 71 doesn't check the interface type, it assumes what interface based on an ifdef...  I think there's a bug here.